docs/pywbemcli/cmdlineinterface.rst

```diff
@@ -529,3 +581,174 @@ Pywbemcli terminates with one of the following program exit codes:
+pywbemcli to get the script itself. It can slow down terminal startup because
+pywbemcli must be called on each terminal startup to get the completion script
+definition
+
+Activation by creating a completion script
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+An alternative is to create a completion script using the same basch statement
+ (ex. ``_PYWBEMCLI_COMPLETE=bash_source pywbemcli``) but
+saving the resulting script in a file.  This that can be executed when
+pywbemcli is to be used and its location is known to create a completion script
+file in a known location for pywbemcli. A file containing shell commands is
+created that defines the completion and completion activation and the tab
+compietion is then activated by executing it (``source ~/.pywbemcli-complete.bash``).
+
+.. table:: Creation of the completion script as the tab completion initialization
+   :name: Shell completion script
+
+  =====  ==========================================================================
+  Shell  Script
+  =====  ==========================================================================
+  bash   _PYWBEMCLI_COMPLETE=bash_source pywbemcli > ~/.pywbemcli-complete.bash
+  zsh    _PYWBEMCLI_COMPLETE=zsh_source pywbemcli > ~/.pywbemcli-complete.zsh
+  fish   _PYWBEMCLI_COMPLETE=fish_source pywbemcli > ~/.config/fish/completions/pywbemcli.fish
+  =====  ==========================================================================
+
+The activation of the completion script can be done either by including the
+activation statement in a terminal startup script (ex. .bashrc) or simply by
+executing the script itself(ex. ``source  ~/.pywbemcli-complete.bash``).
+
+Thus, for bash, pywbemcli can be activated by inserting the following
+evaluation script into .bashrc if pywbemcli is executable whenever the terminal
+is started:
+
+.. code-block::
+
+    eval "$(_PYWBEMCLI_COMPLETE=bash_source pywbemcli)"
+
+or by creating a completion file as follows:
+
+.. code-block::
+
+    PYWBEMCLII_COMPLETE=bash_source pywbemcli > ~/.pywbemcli-complete.bash
+
+and the sourcing that file again when pywbemcli is executable, for example
+as part of the activation of the virtual environment:
+
+.. code-block::
+
+    source  ~/.pywbemcli-complete.bash
+
+Tab completion is activated for the shell by sourcing the completion file when the
+location of pywbemcli is public, for example as follows:
+
+.. code-block:: text
+
+    $ source  ~/.pywbemcli-complete.bash
+
+Testing that pywbemcli tab_completion is activated
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The activation can be tested in a terminal by simply executing a known command
+and using the <TAB> to request completion.  The example below shows testing:
+
+.. code-block:: text
+   $ pywbemcli clas<TAB>
+
+   This should complete the class statement (i.e. expand cmd line to
+   ``pywbemcli class``).
+
+In addition each shell has tools to communicate with the tab-completion
```

Comment:
Why does the user need to know that?

Reply:
Which part above??? DISCUSSION

Reply:
This is about the entire paragraph starting with this line.

Reply:
Largely because I found that if you were having problems, asking bash what completions it knew was useful.  Since we know exactly what should be returned.   And believe me, when it gets to zsh it gets even messier.

Reply:
Discussion

Reply:
Ok but then let's rephrase that to explain how to list conpletions and what to expect for pywbem. As written now, it is not clear to me what it wants to tell me

Reply:
Clean it up a bit.